src/gradio/app/main.py

```diff
@@ -20,40 +20,54 @@
 app = FastAPI()
 
 
-async def process_pdf_and_summarize(file_content, session_id) -> str:
-    logging.info(f"Starting PDF processing for session {session_id}...")
-
-    # TODO: Change return of error messages by raising exception
+def validate_pdf_file(file_content):
     if not file_content:
-        logging.error("No file was uploaded.")
-        return "Error: No file was uploaded."
+        raise HTTPException(status_code=400, detail="No file was uploaded.")
+    if not getattr(file_content, "name", "").lower().endswith(".pdf"):
+        raise HTTPException(status_code=400, detail="Uploaded file is not a PDF.")
 
-    elif not getattr(file_content, "name", "").lower().endswith(".pdf"):
-        logging.error("Uploaded file is not a PDF.")
-        return "Error: The uploaded file is not a PDF."
 
-    else:
+def extract_text_from_pdf(file_content):
+    try:
         pdf_document = pymupdf.open(file_content, filetype="pdf")
         text = ""
         for page_number in range(pdf_document.page_count):
             page = pdf_document.load_page(page_number)
             text += page.get_text("text")
         pdf_document.close()
+        return text
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Error processing the PDF file: {str(e)}"
+        )
 
-        try:
-            loop = asyncio.get_running_loop()
-            summary_task = loop.run_in_executor(None, summarize_text, text)
-            summary_tasks[session_id] = summary_task
-            output_text = await summary_task
-            logging.info("Summarization completed.")
-            return output_text
-        except asyncio.CancelledError:
-            logging.warning(f"Summarization was cancelled for session {session_id}.")
-            return "Summarization processing was stopped by the user."
-        finally:
-            summary_tasks.pop(session_id, None)
+
+async def process_pdf_and_summarize(file_content, session_id) -> str:
+    global summary_tasks
+    logging.info(f"Starting PDF processing for session {session_id}...")
+
+    validate_pdf_file(file_content)  # validation input
+
+    text = extract_text_from_pdf(file_content)  # extract text from pdf
+
+    try:
+        loop = asyncio.get_running_loop()
+        task = loop.run_in_executor(None, summarize_text, text)
+        summary_tasks[session_id] = task
+        output_text = await task
+        logging.info("Summarization completed.")
+        return output_text
+    except Exception as e:
+        logging.error(f"Failed during processing: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to process PDF: {str(e)}")
+    except asyncio.CancelledError:
+        logging.warning(f"Summarization was cancelled for session {session_id}.")
+        raise HTTPException(
```

Comment:
Replace raising error by either `gr.Info("Summarization was cancelled")` or just simply omit the exception

Reply:
`gr.info()` does not exist and cannot be used to inform the user. Do you mean print asyncio.CancelledError?